### PR TITLE
test pahole every 1h, cache result based on pahole sha

### PR DESCRIPTION
**Pull request by tsipa**

### CODE REVIEW

File: travis-ci/vmtest/build_pahole.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ travis_fold start build_pahole "Building pahole"
 CWD=$(pwd)
 REPO_PATH=$1
 PAHOLE_ORIGIN=https://git.kernel.org/pub/scm/devel/pahole/pahole.git
+PAHOLE_BRANCH=${PAHOLE_BRANCH:-master}
```
Comment (anakryiko) [inline travis-ci/vmtest/build_pahole.sh:12]: you probably meant to use $(PAHOLE_BRANCH} below in `git checkout master`, right?

File: .github/workflows/pahole.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        run: exit `cat last_tested_pahole`
+        if: steps.latest.outputs.cache-hit == 'true'
+      - uses: ./.github/actions/vmtest
+        with:
+          kernel: LATEST
+          pahole: $STAGING
+        if: steps.latest.outputs.cache-hit != 'true'
+      - name: Save success
+        run: echo 0 > last_tested_pahole
+        if: steps.latest.outputs.cache-hit != 'true'
```
Comment (anakryiko) [inline .github/workflows/pahole.yml:37]: do I understand correctly that we'll poll latest tmp.master SHA every hour, and will run full test only if it changes from the last time, right? That's pretty nice!

**Review by anakryiko — APPROVED** [review]

**Comment by anakryiko** [comment]
cc @acmel

File: .github/workflows/pahole.yml (hunk 2 above)
Comment (tsipa) [reply .github/workflows/pahole.yml:37]: Yes, that is how it supposed to work.

**Comment by tsipa** [comment]
I think there is some edge cases for canceled jobs and cache and i've read reports about infra issues with cache upload, but i think it's good enough for now and we can polish it later once we discover these edge cases naturally.

**Review by anakryiko — APPROVED** [review]
Awesome, thanks!

**Comment by acmel** [comment]
@tsipa Hey, how can I get the "Run workflow" button to manually trigger https://github.com/libbpf/libbpf/actions/workflows/ondemand.yml ?

**Comment by anakryiko** [comment]
@acmel I've added you to libbpf repo with "Triage" access, please double check if that enables that "Run workflow" button for you?

**Comment by anakryiko** [comment]
@thefallentree, any idea what's missing? What level of access does Arnaldo need to be able to run on-demand workflows?

**Comment by thefallentree** [comment]
Per
https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
, it requires write access to the repo, which my account also doesn't have.

**Comment by anakryiko** [comment]
@acmel, you should be able to fork libbpf/libbpf repo and run workflows on your fork. Can you please try that? I might need to approve it just the first time, not sure. Github UI is a bit confusing and unclear.

**Pull request merged**